addons/point_of_sale/static/src/js/models.js

```diff
@@ -222,10 +222,46 @@ class PosGlobalState extends PosModel {
         this.db.add_partners(this.partners);
     }
     _loadProductProduct(products) {
+        const productMap = {};
+        const productTemplateMap = {};
+
+        function assignApplicableItems(pricelist, correspondingProduct, pricelistItem) {
+            if (!(pricelist.id in correspondingProduct.applicable_pricelists)) {
+                correspondingProduct.applicable_pricelists[pricelist.id] = [];
+            }
+            correspondingProduct.applicable_pricelists[pricelist.id].push(pricelistItem);
+        }
         const modelProducts = products.map(product => {
             product.pos = this;
+            product.applicable_pricelists = {};
```

Comment:
I think you have the pricelist_id on the pricelistItem, so no need to have a dictinnary with the pricelist_id as key, just an array of pricelistItem and when filter later filter the pricelist id

Reply:
indeed.
However by doing so the `get_price` (on the test case does take 8 ms (instead of 2ms with the previous version of the commit). Here the client have 3 pricelists (with ~16 000 items in each) so it's bearable. But if a client have a lot of pricelists, it will take more time linearly.
I did the change the code according to what I understood that you ask, however, personally I do prefer the previous version with the dictionary on the pricelist id.
@caburj may I also have your opinion on that ?

Reply:
@caburj My opinion? I agree with you. I also prefer the previous solution. But you know, we can think of the `applicable_pricelists` array like a constructed many2many relationship with products.
BTW, for this second solution, you'll need to rename `applicable_pricelists` to `applicablePricelistItems`.